src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublishingService.cs

```diff
@@ -61,7 +61,7 @@ public class PublishingService : IPublishingService
                         RetryStageQueue, new RetryStageMessage(releaseId, stage));
 
                     _logger.LogTrace($"Sent retry stage message for Release: {releaseId}");
-                    return new Either<ActionResult, bool>(true);
+                    return new Either<ActionResult, Unit>(Unit.Instance);
```

Comment:
Just return `Unit.Instance`, it automatically gets implicitly converted to an Either

Reply:
I already tried that but OnSuccess matches multiple candiate methods as soon as you return both the left type and right type within the method.

For example, this is ok:

```
        public async Task<Either<ActionResult, Unit>> Test(Guid id)
        {
            return await _persistenceHelper.CheckEntityExists<Release>(id)
                .OnSuccess(async release =>
                {
                    if (false)
                    {
                      // This is ok
                      return Unit.Instance;
                    }

                    return Unit.Instance;
                });
        }
```

but this is not:

```
        public async Task<Either<ActionResult, Unit>> Test(Guid id)
        {
            return await _persistenceHelper.CheckEntityExists<Release>(id)
                .OnSuccess(async release =>
                {
                    if (false)
                    {
                      // This is not ok
                      return ValidationActionResult(ReleaseNotApproved);
                    }

                    return Unit.Instance;
                });
        }
```

Reply:
Eh? That's pretty weird. I'm not sure I really like the whole implicit conversion thing too much when stuff like this happens.

Reply:
I don't understand it but it's not the first time I've run into this problem. Any ideas?

If the return was a method like this it would be fine too

```
        public async Task<Either<ActionResult, Unit>> Test(Guid id)
        {
            return await _persistenceHelper.CheckEntityExists<Release>(id)
                .OnSuccess(async release =>
                {
                    if (false)
                    {
                        return ValidationActionResult(ReleaseNotApproved);
                    }
                    
                    return Result();
                });
        }

        public Either<ActionResult, Unit> Result()
        {
            return Unit.Instance;
        }
```

Reply:
No idea unfortunately. This variant makes sense because you're just creating another `Either`, as if you were doing `new Either`, but I would have expected the conversion to just work with `Unit.Instance`. Just testing it in other places, I can do exactly that and it works!